sympy/assumptions/ask.py

```diff
@@ -9,66 +9,1114 @@
+        >>> x = Symbol('x')
+        >>> ask(Q.complex(0))
+        True
+        >>> ask(Q.complex(2 + 3*I))
+        True
+        >>> ask(Q.complex(oo))
+        False
+
+        References
+        ==========
+
+        .. [1] https://en.wikipedia.org/wiki/Complex_number
+
+        """
+        return Predicate('complex')
+
+    @memoize_property(predicate_storage)
+    def algebraic(self):
+        """
+        Algebraic number predicate.
+
+        ``Q.algebraic(x)`` is true iff ``x`` belongs to the set of
+        algebraic numbers.
+
+        Examples
+        ========
+
+        >>> from sympy import ask, Q, sqrt, I, pi
+        >>> ask(Q.algebraic(sqrt(2)))
+        True
+        >>> ask(Q.algebraic(I))
+        True
+        >>> ask(Q.algebraic(pi))
+        False
+
+        References
+        ==========
+
+        .. [1] http://en.wikipedia.org/wiki/Algebraic_number
+        """
+        return Predicate('algebraic')
+
+    @memoize_property(predicate_storage)
+    def transcendental(self):
+        """
+        Transcedental number predicate.
+
+        ``Q.algebraic(x)`` is true iff ``x`` belongs to the set of
+        transcendental numbers. A transcendental number is a real
+        or complex number that is not algebraic.
+
+        TODO: Add examples
+
+        """
+        return Predicate('transcendental')
+
+    @memoize_property(predicate_storage)
+    def integer(self):
+        """
+        Integer predicate.
+
+        ``Q.integer(x)`` is true iff ``x`` belongs to the set of integer numbers.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask, S
+        >>> ask(Q.integer(5))
+        True
+        >>> ask(Q.integer(S(1)/2))
+        False
+
+        References
+        ==========
+
+        .. [1] https://en.wikipedia.org/wiki/Integer
+
+        """
+        return Predicate('integer')
+
+    @memoize_property(predicate_storage)
+    def rational(self):
+        """
+        Rational number predicate.
+
+        ``Q.rational(x)`` is true iff ``x`` belongs to the set of
+        rational numbers.
+
+        Examples
+        ========
+
+        >>> from sympy import ask, Q, pi, S
+        >>> ask(Q.rational(0))
+        True
+        >>> ask(Q.rational(S(1)/2))
+        True
+        >>> ask(Q.rational(pi))
+        False
+
+        References
+        ==========
+
+        https://en.wikipedia.org/wiki/Rational_number
+
+        """
+        return Predicate('rational')
+
+    @memoize_property(predicate_storage)
+    def irrational(self):
+        """
+        Irrational number predicate.
+
+        ``Q.irrational(x)`` is true iff ``x``  is any real number that
+        cannot be expressed as a ratio of integers.
+
+        Examples
+        ========
+
+        >>> from sympy import ask, Q, pi, S
+        >>> ask(Q.irrational(0))
+        False
+        >>> ask(Q.irrational(S(1)/2))
+        False
+        >>> ask(Q.irrational(pi))
+        True
```

Comment:
Probably should also show something like `ask(Q.irrational(I))` to drive home the "real" part of the definition.